engine/flink/components/base/src/main/scala/pl/touk/nussknacker/engine/flink/util/transformer/aggregate/aggregates.scala

```diff
@@ -171,6 +172,84 @@ object aggregates {
 
   }
 
+  object CountWhenAggregator extends Aggregator {
+
+    override type Element = java.lang.Boolean
+
+    override type Aggregate = java.lang.Long
+
+    override def zero: java.lang.Long = 0L
+
+    override def addElement(element: java.lang.Boolean, aggregate: Aggregate): Aggregate =
+      if (element) aggregate + 1 else aggregate
+
+    override def mergeAggregates(aggregate1: Aggregate, aggregate2: Aggregate): Aggregate = aggregate1 + aggregate2
+
+    override def result(finalAggregate: Aggregate): AnyRef = finalAggregate
+
+    override def computeOutputType(input: typing.TypingResult): Validated[String, typing.TypingResult] = {
+      if (input.canBeSubclassOf(Typed[Boolean])) {
+        Valid(Typed[Long])
+      } else {
+        Invalid(s"Invalid aggregate type: ${input.display}, should be: ${Typed[Boolean].display}")
+      }
+    }
+
+    override def computeStoredType(input: typing.TypingResult): Validated[String, typing.TypingResult] =
+      computeOutputType(input)
+  }
+
+  object AverageAggregator extends Aggregator {
+
+    override type Element = java.lang.Number
+
+    override type Aggregate = AverageAggregatorState
+
+    override def zero: AverageAggregatorState = AverageAggregatorState(null, 0)
+
+    override def addElement(element: java.lang.Number, aggregate: Aggregate): Aggregate =
+      AverageAggregatorState(MathUtils.largeFloatingSum(element, aggregate.sum), aggregate.count + 1)
+
+    override def mergeAggregates(aggregate1: Aggregate, aggregate2: Aggregate): Aggregate =
+      AverageAggregatorState(
+        MathUtils.largeFloatingSum(aggregate1.sum, aggregate2.sum),
+        aggregate1.count + aggregate2.count
+      )
+
+    override def result(finalAggregate: Aggregate): AnyRef = {
+      val count = finalAggregate.count
+      finalAggregate.sum match {
+        case null =>
+          // will be replaced to Double.Nan in alignToExpectedType iff return type is known to be Double
+          null
+        case sum: java.lang.Double     => Predef.double2Double(sum / count)
+        case sum: java.math.BigDecimal => (BigDecimal(sum) / BigDecimal(count)).bigDecimal
+      }
+    }
+
+    override def alignToExpectedType(value: AnyRef, outputType: TypingResult): AnyRef = {
+      if (value == null && outputType == Typed(classOf[lang.Double])) {
```

Comment:
It can be just `Typed[Double]`, we always convert primitives to wrappers in `Typed.typedClass`. Thanks to that we can use scala idiomatic types